:lipstick: pretty up the props box on aksel.nav.no + typos

JulianNymark opened the pull request:
- now allows you to write markdown in the JSDoc.
- plain URLs -> links require the remark-gfm plugin
- uses our own <InlineCode> component for markdown parsing tag 'code'.
- uses our own link component (via a wrapper since there's no link text in plain URL links)

JulianNymark committed:
:lipstick: pretty up the props box on aksel.nav.no + typos

- also update some props docs here & there

JulianNymark committed:
:recycle: pull out component, lazy() it

JulianNymark committed:
:fire: remove testing css

JulianNymark committed:
:art: add fallback with <Suspense>

JulianNymark committed:
:speech_balloon: correct @default strings

JulianNymark committed:
:lipstick: spacing in Props table

JulianNymark committed:
:fire: @note is not a real JSDoc tag

JulianNymark committed:
:sparkles: docgen extracts example, render fancy example box

JulianNymark committed:
:sparkles: pretty @return and @params

JulianNymark committed:
:lipstick: improve table layout / CSS

JulianNymark committed:
:lipstick: + update CopyButton & Combobox jsdoc.

JulianNymark committed:
:memo: update jsdoc

KenAJoh committed:
:art: Oppdatert layout og visning av prop-table

JulianNymark committed:
:art: use more semantic html, better nested <dl>

JulianNymark committed:
:wheelchair: add h4 to prop items

KenAJoh commented on aksel.nav.no/website/sanity/config.ts:
```diff
@@ -1,6 +1,6 @@
 export const SANITY_PROJECT_ID = "hnbe3yhs";
 export const SANITY_API_VERSION = "2021-10-21";
-export const SANITY_DATASET = "production";
+export const SANITY_DATASET = "development";
```
Obs, må endres tilbake til production før merge ⚠️ ⚠️

JulianNymark committed:
:construction_worker: use production sanity data

JulianNymark committed:
:wheelchair: revert nested <dl> so Axe tests pass

JulianNymark committed:
:label: correct some types for tsc to pass

JulianNymark committed:
:wheelchair: change from div -> ul

HalvorHaugan commented on aksel.nav.no/website/package.json:
Ser ut som du ikke har kjørt `yarn` og committet lockfilen

JulianNymark replied:
hmm, trodde det ikke var endret på noen deps, så da pleier jeg å ikke pushe den 🤔. lurer på om det er når jeg bytter branches at lockfile blir ut av sync og da generer nye checksums.

JulianNymark replied:
men her la jeg jo til nye pakker 🙈

JulianNymark committed:
:construction_worker: commit lockfile

HalvorHaugan commented on aksel.nav.no/website/components/sanity-modules/props/parts/PropTabell.tsx:
```diff
@@ -28,22 +28,42 @@ const PropTable = ({ komponent }: PropTableProps) => {
-              parent={komponent?.title ?? ""}
-            />
+            <div className="border border-t-0 border-gray-300" key={prop.name}>
+              <dt className="px-2 py-2">
```
Synes det ble litt mye spacing mellom prop-navn og "Type:". Kanskje bottom padding kan sløyfes her?
```suggestion
              <dt className="px-2 pt-2">
```

HalvorHaugan commented on aksel.nav.no/website/components/sanity-modules/props/parts/DtList.tsx:
```diff
@@ -1,48 +1,63 @@
-      {prop.description && (
-        <dd className="whitespace-pre-wrap text-base">{prop.description}</dd>
+      {prop.type && (
+        <li className="my-4 flex px-3 text-base">
```
Er pirkete nå, men synes godt vi kan gå ned til `my-3` på disse radene.

HalvorHaugan commented on aksel.nav.no/website/components/sanity-modules/props/parts/DtList.tsx:
```diff
@@ -1,48 +1,63 @@
-          )}`}
-        </dd>
+      {prop.defaultValue && (
+        <li className="flex px-3 text-base">
```
Ser ut som det mangler block margin på "Default"

HalvorHaugan commented on aksel.nav.no/website/components/sanity-modules/props/parts/DtList.tsx:
```diff
@@ -1,48 +1,63 @@
+      {prop.type && (
+        <li className="my-4 flex px-3 text-base">
+          <div className="min-w-24 font-semibold">Type: </div>
+          <code className="text-sm">{Highlighter({ type: prop.type })}</code>
```
Skulle gjerne sett at type-raden var bedre aligned. Hva med å legge på `mt-05` på `<code>`? 😇
```suggestion
          <code className="mt-05 text-sm">{Highlighter({ type: prop.type })}</code>
```
En annen mulighet er å gi de samme line-height.

JulianNymark committed:
:lipstick: make ul bottom rounded and align type with it\'s code block

HalvorHaugan commented on @navikt/core/react/src/alert/Alert.tsx:
```diff
@@ -12,37 +12,39 @@ import { BodyLong } from "../typography";
-   * requires closeButton to be true
+   * Callback for alert wanting to close.
+   *
+   * **requires closeButton to be true**.
```
```suggestion
   * **Requires closeButton to be true**.
```

HalvorHaugan commented on @navikt/core/react/src/layout/stack/Stack.tsx:
```diff
@@ -7,7 +7,11 @@ import { ResponsiveProp, SpacingScale } from "../utilities/types";
+   *
+   * @example
+   * gap='center'
+   * gap={{xs: 'start', sm: 'center', md: 'end', lg: 'space-around', xl: 'space-between'}}
```
```suggestion
   * justify='center'
   * justify={{xs: 'start', sm: 'center', md: 'end', lg: 'space-around', xl: 'space-between'}}
```

HalvorHaugan commented on @navikt/core/react/src/layout/stack/Stack.tsx:
```diff
@@ -18,21 +22,31 @@ export interface StackProps extends HTMLAttributes<HTMLDivElement> {
+   *
+   * @example
+   * gap='center'
+   * gap={{xs: 'start', sm: 'center', md: 'end', lg: 'baseline', xl: 'stretch'}}
```
```suggestion
   * align='center'
   * align={{xs: 'start', sm: 'center', md: 'end', lg: 'baseline', xl: 'stretch'}}
```

HalvorHaugan commented on @navikt/core/react/src/layout/stack/Stack.tsx:
```diff
@@ -18,21 +22,31 @@ export interface StackProps extends HTMLAttributes<HTMLDivElement> {
+   * gap=true
+   * gap={{xs: true, sm: false, md: true, lg: false, xl: true}}
    */
   wrap?: boolean;
```
Dette er bare en boolean. Trenger kanskje ikke eksempler i det hele tatt?

HalvorHaugan commented on @navikt/core/react/src/loader/Loader.tsx:
Linken i @see rendres to ganger
![image](https://github.com/navikt/aksel/assets/83693529/4cde27c1-f001-4aa2-bb17-136a0f2015d7)

HalvorHaugan commented on @navikt/core/react/src/modal/types.ts:
```diff
@@ -1,25 +1,25 @@
   /**
-   * Content for the header. Alteratively you can use <Modal.Header> instead for more control,
+   * Content for the header. Alteratively you can use `<Modal.Header>` instead for more control,
    * but then you have to set `aria-label` or `aria-labelledby` on the modal manually.
```
Her ble det linjeskift på et uheldig sted. Kanskje bare slå sammen til én linje?
![image](https://github.com/navikt/aksel/assets/83693529/5e0a3e41-16b7-45f4-92d6-65caff5a88b9)

HalvorHaugan commented on aksel.nav.no/website/components/sanity-modules/props/parts/DtListDescription.tsx:
```diff
@@ -0,0 +1,26 @@
+        remarkPlugins={[remarkGfm]}
+        components={{
+          code: InlineCode,
+          a: LinkWrapper as any,
```
Prøv å type children

HalvorHaugan commented on aksel.nav.no/website/components/sanity-modules/props/parts/DtListExample.tsx:
```diff
@@ -0,0 +1,11 @@
+  );
+};
+
+export default DtListDescription;
```
```suggestion
export default DtListExample;
```

HalvorHaugan commented on scripts/docgen.ts:
```diff
@@ -19,6 +19,44 @@ const options: docgen.ParserOptions = {
+const enrich_extra_prop_fields = (docs: docgen.ComponentDoc[]) => {
+  for (const doc of docs) {
+    for (const prop of Object.values(doc.props)) {
+      if (prop.description) {
```
Vurder "early return" (`continue`) for å redusere indentation.

HalvorHaugan commented on aksel.nav.no/website/components/sanity-modules/props/parts/DtList.tsx:
```diff
@@ -1,48 +1,63 @@
+  loading: () => <Skeleton />,
+});
+
+export const DtList = ({ prop }: { prop: any; parent: string }) => {
```
Kan vi unngå `any` her?

JulianNymark committed:
:wheelchair: set language tag for props written in english

JulianNymark committed:
:label: type improvements for props section

JulianNymark committed:
:see_no_evil: update .gitignore

JulianNymark committed:
:memo: jsdoc update

JulianNymark committed:
:lipstick: props section

JulianNymark committed:
:recycle: rename and less indentation (continues > ifs)

JulianNymark committed:
:lipstick: style changes for Prop section

- breakpoint fail fix
- breakpoint col / row flex for prop fields
- padding tweaks

JulianNymark committed:
:memo: update example jsdoc
